subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -63,8 +63,8 @@
 
 #include "hci_internal.h"
 
-static K_SEM_DEFINE(sem_prio_recv, 0, K_SEM_MAX_LIMIT);
-static K_FIFO_DEFINE(recv_fifo);
+static struct k_sem sem_prio_recv;
+static struct k_fifo recv_fifo;
```

Comment:
Why were the static initializers removed in favor of `k_fifo_init` and `k_sem_init`?

Reply:
Initializing a FIFO makes it empty. Initializing a semaphore resets its count to 0. Since the HCI driver `close()` function can be called at any time, we do need to reset these structures every time `open()` is called (or once at boot and then every time `close()` is called.

>Does it work as intended if hci_driver_open is called multiple times? 

You are not supposed to call `open()` multiple times without calling `close()` first. But if you do that, this will make it work as intended, since we reset all the structs at that point.